components/bt/host/bluedroid/api/include/api/esp_gap_ble_api.h

```diff
@@ -17,2687 +17,3984 @@
+    ESP_BLE_AD_TYPE_128SOL_SRV_UUID          = 0x15,    /*!< List of 128-bit service solicitation UUIDs. */
+    ESP_BLE_AD_TYPE_SERVICE_DATA             = 0x16,    /*!< Service data. */
+    ESP_BLE_AD_TYPE_PUBLIC_TARGET            = 0x17,    /*!< Public target address. */
+    ESP_BLE_AD_TYPE_RANDOM_TARGET            = 0x18,    /*!< Random target address. */
+    ESP_BLE_AD_TYPE_APPEARANCE               = 0x19,    /*!< Appearance. */
+    ESP_BLE_AD_TYPE_ADV_INT                  = 0x1A,    /*!< Advertising interval. */
+    ESP_BLE_AD_TYPE_LE_DEV_ADDR              = 0x1b,    /*!< Bluetooth LE device address. */
+    ESP_BLE_AD_TYPE_LE_ROLE                  = 0x1c,    /*!< Bluetooth LE role. */
+    ESP_BLE_AD_TYPE_SPAIR_C256               = 0x1d,    /*!< Pairing data, C256. */
+    ESP_BLE_AD_TYPE_SPAIR_R256               = 0x1e,    /*!< Pairing data, R256. */
+    ESP_BLE_AD_TYPE_32SOL_SRV_UUID           = 0x1f,    /*!< List of 32-bit service solicitation UUIDs. */
+    ESP_BLE_AD_TYPE_32SERVICE_DATA           = 0x20,    /*!< 32-bit service data. */
+    ESP_BLE_AD_TYPE_128SERVICE_DATA          = 0x21,    /*!< 128-bit service data. */
+    ESP_BLE_AD_TYPE_LE_SECURE_CONFIRM        = 0x22,    /*!< Bluetooth LE secure connections confirmation value. */
+    ESP_BLE_AD_TYPE_LE_SECURE_RANDOM         = 0x23,    /*!< Bluetooth LE secure connections random value. */
+    ESP_BLE_AD_TYPE_URI                      = 0x24,    /*!< Uniform Resource Identifier. */
+    ESP_BLE_AD_TYPE_INDOOR_POSITION          = 0x25,    /*!< Indoor positioning. */
+    ESP_BLE_AD_TYPE_TRANS_DISC_DATA          = 0x26,    /*!< Transport discovery data. */
+    ESP_BLE_AD_TYPE_LE_SUPPORT_FEATURE       = 0x27,    /*!< Bluetooth LE supported features. */
+    ESP_BLE_AD_TYPE_CHAN_MAP_UPDATE          = 0x28,    /*!< Channel map update indication. */
+    ESP_BLE_AD_MANUFACTURER_SPECIFIC_TYPE    = 0xFF,    /*!< Manufacturer specific data. */
 } esp_ble_adv_data_type;
 
-/// Advertising mode
+/**
+ * @brief Advertising type
+ *        - This enum represents the different types of Bluetooth Low Energy (LE) advertising types. Each types dictates how a device advertises its presence and availability for connections.
+ *        - More detail in spec: BLUETOOTH CORE SPECIFICATION Version 5.4 Vol 4, Part E and BLUETOOTH CORE SPECIFICATION Version 5.4 Vol 6, Part B.
+ */
 typedef enum {
-    ADV_TYPE_IND                = 0x00,
-    ADV_TYPE_DIRECT_IND_HIGH    = 0x01,
-    ADV_TYPE_SCAN_IND           = 0x02,
-    ADV_TYPE_NONCONN_IND        = 0x03,
-    ADV_TYPE_DIRECT_IND_LOW     = 0x04,
+    ADV_TYPE_IND                = 0x00, /*!< Connectable and scannable undirected advertising (ADV_IND).
+                                             - This is a general advertising type used when the device can be connected by other target. */
+    ADV_TYPE_DIRECT_IND_HIGH    = 0x01, /*!< Connectable high duty cycle directed advertising (ADV_DIRECT_IND, high duty cycle).
```

Comment:
The connectable directed advertising event type allows an initiator to respond
so that both the advertiser and initiator will enter the Connection state.